test/gateway_composer/revoke_stake_request.js

```diff
@@ -0,0 +1,180 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const BN = require('bn.js');
+const utils = require('../test_lib/utils');
+const { AccountProvider } = require('../test_lib/utils.js');
+const web3 = require('../test_lib/web3.js');
+
+const gatewayComposerUtils = require('./utils');
+
+contract('GatewayComposer::revokeStakeRequest', async (accounts) => {
+    describe('Negative Tests', async () => {
+        const accountProvider = new AccountProvider(accounts);
+
+        it('Fails when owner is not the caller.', async () => {
+            const {
+                gatewayComposer,
+            } = await gatewayComposerUtils.setupGatewayComposer(accountProvider);
+
+            const stakeRequestHash = web3.utils.soliditySha3('hash');
+            utils.expectRevert(gatewayComposer.revokeStakeRequest(
+                stakeRequestHash,
+                { from: accountProvider.get() },
+            ),
+            'Should revert as msg.sender is not the owner.',
+            'Only owner can call the function.');
+        });
+
+        it('Fails when requestHash is invalid.', async () => {
```

Comment:
Suggest revising to, "Fails when stake request is not found", because that's what this test is actually checking for; in addition, an empty string is not _per se_ invalid and the tested function does validate the content of the hash.